ext/standard/php_uuid.h

```diff
@@ -0,0 +1,563 @@
+ *
+ * Versions 1 and 2 are not supported due to privacy/security concerns. Refer
+ * to the Wikipedia article for more information.
+ *
+ * This implementation is inspired by many other implementations:
+ * - [RFC 4122 Sample Implementation](https://tools.ietf.org/html/rfc4122#appendix-A)
+ * - [Rust UUID](https://github.com/rust-lang-nursery/uuid)
+ * - [util-linux LibUUID](https://github.com/karelzak/util-linux)
+ * - [boost Uuid](http://www.boost.org/doc/libs/1_64_0/libs/uuid/)
+ * - [D std.uuid](https://dlang.org/library/std/uuid.html)
+ *
+ * ## Examples
+ * ```
+ * php_uuid uuid;
+ * 
+ * // A random UUID with more randomness than the version 4 implementation.
+ * // This should NOT be used in real-world applications!
+ * php_random_bytes_silent(&uuid, PHP_UUID_LEN);
+ * php_uuid_set_variant(&uuid, PHP_UUID_VARIANT_FUTURE_RESERVED);
+ * 
+ * php_uuid_create_v3(&uuid, &PHP_UUID_NAMESPACE_DNS, "php.net", sizeof("php.net") - 1);
+ * assert(php_uuid_is_nil(uuid) == FALSE);
+ * assert(php_uuid_get_variant(uuid) == PHP_UUID_VARIANT_RFC4122);
+ * assert(php_uuid_get_version(uuid) == PHP_UUID_VERSION_3_NAME_BASED_MD5);
+ * 
+ * php_uuid_create_v4_silent(&uuid);
+ * assert(php_uuid_is_nil(uuid) == FALSE);
+ * assert(php_uuid_get_variant(uuid) == PHP_UUID_VARIANT_RFC4122);
+ * assert(php_uuid_get_version(uuid) == PHP_UUID_VERSION_4_RANDOM);
+ * 
+ * php_uuid_create_v5(&uuid, &PHP_UUID_NAMESPACE_DNS, "php.net", sizeof("php.net") - 1);
+ * assert(php_uuid_is_nil(uuid) == FALSE);
+ * assert(php_uuid_get_variant(uuid) == PHP_UUID_VARIANT_RFC4122);
+ * assert(php_uuid_get_version(uuid) == PHP_UUID_VERSION_5_NAME_BASED_SHA1);
+ * 
+ * php_uuid_parse_silent(&uuid, "urn:uuid:123E4567-E89B-12D3-A456-426655440000", sizeof("urn:uuid:123E4567-E89b-12D3-A456-426655440000") - 1);
+ * assert(php_uuid_is_nil(uuid) == FALSE);
+ * assert(php_uuid_get_variant(uuid) == PHP_UUID_VARIANT_RFC4122);
+ * assert(php_uuid_get_version(uuid) == PHP_UUID_VERSION_1_TIME_BASED);
+ * 
+ * assert(memcmp(&uuid, "\x12\x3E\x45\x67\xE8\x9B\x12\xD3\xA4\x56\x42\x66\x55\x44\x00\x00", PHP_UUID_LEN) == 0);
+ *
+ * php_uuid_hex hex;
+ * php_uuid_to_hex(&hex, uuid);
+ * assert(memcmp(&hex, "123e4567e89b12d3a456426655440000", PHP_UUID_HEX_LEN) == 0);
+ * 
+ * php_uuid_string str;
+ * php_uuid_to_string(&str, uuid);
+ * assert(memcmp(&str, "123e4567-e89b-12d3-a456-426655440000", PHP_UUID_STRING_LEN) == 0);
+ * ```
+ *
+ * @see https://en.wikipedia.org/wiki/Universally_unique_identifier
+ * @see https://tools.ietf.org/html/rfc4122
+ */
+
+#ifndef PHP_UUID_H
+#define PHP_UUID_H
+
+#include "zend_types.h"
+#include "php.h"
+
+/** UUID class entry for usage by other modules. */
+PHPAPI extern zend_class_entry *php_ce_UUID;
+
+/** UUIDParseException class entry for usage by other modules. */
+PHPAPI extern zend_class_entry *php_ce_UUIDParseException;
+
+/** UUID binary representation to store the 128 bit number in 16 bytes. */
+PHPAPI typedef uint8_t php_uuid[16];
```

Comment:
I would recommend against typedef'ing array types. They have confusing passing semantics (by-value in declarations, by-pointer in arguments) and in the argument case the length is lost entirely. 

If you want to have separate types for these, please use a struct wrapper.